bpf/kmesh/workload/include/service.h

```diff
@@ -37,6 +41,47 @@ static inline int lb_random_handle(struct kmesh_context *kmesh_ctx, __u32 servic
     return 0;
 }
 
+static inline int
+lb_locality_failover_handle(struct kmesh_context *kmesh_ctx, __u32 service_id, service_value *service_v, bool is_strict)
+{
+    int ret = 0;
+    uint32_t rand_k = 0;
+    endpoint_key endpoint_k = {0};
+    endpoint_value *endpoint_v = NULL;
+    endpoint_k.service_id = service_id;
+
+    // #pragma unroll
+    for (int match_prio = 0; match_prio < PRIO_COUNT; match_prio++) {
+        endpoint_k.prio = match_prio; // 0->6
+        // if we have endpoints in this prio
+        if (service_v->prio_endpoint_count[match_prio] > 0) {
+            rand_k = bpf_get_prandom_u32() % service_v->prio_endpoint_count[match_prio] + 1;
+            if (rand_k >= MAP_SIZE_OF_BACKEND) {
+                return -ENOENT;
+            }
+            endpoint_k.backend_index = rand_k;
+            endpoint_v = map_lookup_endpoint(&endpoint_k);
+            if (!endpoint_v) {
+                BPF_LOG(
+                    ERR, SERVICE, "find endpoint [%u/%u/%u] failed", service_id, match_prio, endpoint_k.backend_index);
+                return -ENOENT;
+            }
+            ret = endpoint_manager(kmesh_ctx, endpoint_v, service_id, service_v);
+            if (ret != 0) {
+                if (ret != -ENOENT)
+                    BPF_LOG(ERR, SERVICE, "endpoint_manager failed, ret:%d\n", ret);
+                return ret;
+            }
+            return 0; // find the backend successfully
+        }
+        if (is_strict && match_prio == service_v->lb_strict_prio) { // only match lb strict index
+            return -ENOENT;
```

Comment:
This is not align with ztunnel behavior, this return value would make the traffic goes through like directly accessing the clusterIP, but not blocking the traffic